### PR TITLE
Bandit changes p1

### PR DESCRIPTION
**Pull request by Rayforx**
## About The Pull Request
This PR adds the warlock as a bandit class: [Warlock PR by josephpatrickgreen](https://github.com/Tree445/Hearthstone/pull/634) and [Ham-Hole's Almanac of Mayhem by josephpatrickgreen](https://github.com/Tree445/Hearthstone/pull/552) Thank you to the original author for the amazing work

**Warlock changes**
Adds pack weapons, and the framework for it
I added a bunch of new spells that another coder wanted. For now, only the warlock can use them.
Adds warlock and the framework to it (balance might be bad)
This adds the ability for the warlock to choose a being in the world to praise, that gives him various powers. Experiment!
30% chance to roll warlock, 1 role.

**PQ adjustment:**
Adjusted the PQ for Hedge Knight to be 15

**Weapon skills**
Adjusted the weapon skills for the bandits to be 4, 1 above towners, made room for a support player at round start.

**Hoardmaster changes**
Adjusted the price of the health potion to be the cost of pink pills before it was removed, with a price increase (35)
Hoardmaster cleanup, which was badly needed
Removed all the black steel armor, and replaced it with the light gambeson for variety (iron armor)
Moved the hoardmaster file into a bandit folder, and put them into categories based around type and cost
Added short swords to the knave on request

Sprites
added sprites for the various warlock-related items

This PR should be TM'ed.

Otherworldly Patron (Subclasses)

Warlocks choose 1 of 8 patrons on game start:
**archfey (face/caster)**

-Grants most seelie spells
-give wild magic trait (not yet implemented)
**celestial (healer)**

-give eldritch healing
**fathomless (jack of all trades)**

-dodge expert trait
-medium armor trait
-water breathing trait
**fiend (combat caster)**

-no fire trait
-kiss of the fiend trait
**genie (face)**

-signet ring
-rich pouch
-magic lamp (just flavor)
-dodge expert trait
**great old one (utility caster)**

-you gain the learn magic tab
**hexblade (melee)**

-dodge expert trait
-medium armor trait
-pact weapon
**undead (tank)**

-medium armor trait
-heavy armor trait
-no stink trait
-infestation spell
-cloak of flies spell

**#Boon (what did you trade your faith for?)**

This is another round start choice (if the warlock feels underpowered after testing, give them 2)
strength

**-pact weapon**
+1 strength
**knowledge**

-patron grimoire added to hands
+1 intelligence
**love**

-give soulbond ring
-goodlover trait
**power**

-trade eldritch blast for empowered eldritch blast
**health**

-give eldritch healing
+2 athletics
**wealth**

-rich pouch
-give eldritch talisman
purpose

-constellation chain added to wrist
**friendship**

Find Familiar spell
Wild Magic

5% chance for spells to do something random, maybe
**Kiss of the Fiend**

when you cast certain projectile spells, hurl a firebolt in the same direction a half second later.
**Signet Ring**

While worn you have the noble trait
**Pact Weapon**

datum added to a weapon
-1.2x stats
-named
-given "singing item"
-non-owner wielders are burned periodically
-can be right-clicked by owner to turn into different weapon
-can invoke with mmb to prompt a poll for ghosts to possess it
Singing Item

datum added to an item
-random personality: compassionate, bloodthirsty, righteous, manipulative, paranoid, singing
-talks when attacking, wielded or dropped
**Cloak of Flies**

A toggleable spell which creates a cloud of flies that deals damage over time to inhalers
**Patron Grimoire**

A spellbook which when held in hands or placed on hip gives you access to 3 random cantrips and message

**Soulbond Ring**
a ring which when worn by another mob, soul links you but gives the other creature the power to resist magic, this mob will have a heart attack after 30 seconds if you die.

**Empowered Eldritch Blast**

Eldritch blast with one square knockback that has a longer cd and more damage

**Eldritch Healing**

A faithless version of lesser miracle

**Eldritch Talisman**
a talisman which, while worn gives you more fortune
**Constellation Chain**
A chain which, while worn (on wrist) gives you nightvision)
double the force of a regular chain so it may be used as a blunt instrument

**Find Familiar**

A spell that polls ghosts in an attempt to create a hell goblin or skeleton after use, this spell deletes itself from your known spells

## Why It's Good For The Game
There are more classes, a framework for awesome spells in the future, and more awesome mechanics that can be put on weapons, including, the vampire lord.

### CODE REVIEW

**Comment by notafurryy** [comment]
I don't think people like gun roles very much and it may need a small nerf, journeyman swords and wrestling are a little borked letting them get out of most situations and fight up close while still being devastating at range. Also expert firearms is a bit much, could lower it to journeyman, oh also giving them double pistols instead of the handgonne would be fun!

**Comment by notafurryy** [comment]
Oh also, not sure bandits need the skill buffs to expert, thry are more than decently powerful as is, I assume even more so with the new classes

**Comment by Purpleturtle3** [comment]
>Ports Heartstone bloat spells
>Powercreeps all bandits by giving them expert in 4
>Changes hedge knights so they can pick up a far better weapon group than swords (Polearms)
>Roundstart 4 firearm skills (more than most of the roles who have gun skill at all) and armed with a gun bandit

**Comment by john-pawson** [comment]
> Adjusted the weapon skills for the bandits to be 4, 1 above towners, made room for a support player at round start

While i would've liked this change and it would've made sense. All towners got their weapon skills lowered to apprentice or below, not only that, but the refugees are usually weaker than the adventurers we had before so... Yeah, this isn't needed anymore.

>Gunslinger added two roles. It's people with guns.

Oooh yeah. Bad idea. Please don't. 
Guns are supposed to be *very* rare, and i **really** don't think giving the bandits just a class with them is, or will be, a good idea. Considering how they're just an instant artery through anything right now.
And i say this while having in mind that i don't like the idea of the handgonner being a refugee option. But in comparison they get absolutely 0 combat skills other than apprentice knives and wrestling. No armor training. No armor when they spawn. And no other weapons outside of the handgonne.
While the gunslinger gets journeyman on swords too, a bunch of other skills, and a shop that lets him buy new armors and weapons. Have you considered that if you give the bandits the ability to just buy weapons, they could all carry around a gun and fire it without any skill? Considering how the stun from failure delivers very little if any damage to the user and stuns for less time than the one getting shot, them just carrying around a bunch of guns to insta-crit their opponents *isn't* that bad of a strategy. Even more when you can have 3-4 shooting guns and the rest defending them while they wait their 3 seconds stuns to be over. Compared to the 3-6 second stun and artery crit their opponents now have.

**Comment by Rayforx** [comment]
Draft PR, I did see the changes to towners, yeah

tirs. 24. dec. 2024 19.14 skrev JP ***@***.***>:

**Comment by john-pawson** [comment]
> Draft PR, I did see the changes to towners, yeah

Oh nice.

**Comment by Rayforx** [comment]
Also, the guns are not more rare than a crossbow right now, which do mean
that their is no reason that the new class should not be made to support
everyone having guns outside of them. The skills are currently a
placeholder as it was directly taken from the musketeer. Guns being shot by
someone without skills are a death sentence, and a patch is being made to
make sure that people can't buy from each other's panel. Guns are
incredibly expensive in the hoardmaster for the same reason you mentioned,
and the gunslinger can only buy a pistol or arquebus with the usual amount
of triumph.

tirs. 24. dec. 2024 19.17 skrev JP ***@***.***>:

**Comment by Gyrandola** [comment]
The new mechanics and spells are quite cool. Just be aware the faction is in for a major rework and reflavor.

I'll leave the balance review for when this is ready.

**Comment by Koteshkin** [comment]
I like warlock as class, but **Matthiosit** warlock sounds VERY stupid.
Honestly bandits need to be let any patron, but ten. And if someone with ten worship registered bandit, force it to be godless.

**Comment by Rayforx** [comment]
What dupes are you even talking about, I am on the latest branch. If you want to claim that I put in economy dupes that was removed, then you should at least say what they are, so I can go remove them.

The poison arrow was not removed; it was made 50 triumphs back in the day, which is the same as a quiver. They were nerfed in another PR after the poison effect was reduced.

The spells were added for another coder and also, since, the warlock needs them.

I can atomize the gunslinger, but it should not affect a lot of the actual balance of it. 

Skeleton hand is also not used by the warlock, the only place that it would be "available" is through the book of his. Which, I can remove. It's not something you can choose in any other way.

I poked Gyran about the skill level 4 for weapons skills for bandits before I did it. It's their choice if they do not want to respond about it.

Warlock are supposed to be a heavy hitter for bandits, but they do the same as a mage or the mage mercenary.

The talisman being slop is a personal opinion, ngl on that one.

File: code/modules/cargo/packsrogue/bandit/warlock.dm [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+/datum/supply_pack/rogue/bandit/Warlock/steelflail
+	name = "Steel Flail"
+	cost = 150
+	contains = list(/obj/item/rogueweapon/flail/sflail)
+//mace
+
+/datum/supply_pack/rogue/bandit/Warlock/cudgel
+	name = "Cudgel"
+	cost = 30
```
[the remark this answers is not in the document]
Comment (john-pawson) [reply code/modules/cargo/packsrogue/bandit/warlock.dm:105]: To be fair, this one's the cudgel. The mostly wooden weapon that turns to ash if you try to smelt it.

**Comment by ShiningM00N** [comment]
I'm sorry, but giving mages access to medium and **heavy** armors is a balance mistake in the making and a gross type of buff that feels more like "I'm going to make my faction super strong, and also super unique". Even if this was given to Court magos it'd be outrageously strong and just gross.

**Comment by Rayforx** [comment]
From what I can gather, then the consensus was that the class was a bit too strong with some of the classes in the warlock, and had to be balanced, and that it needed a general balance pass.
And the armor needed to be changed to light armor with the max being medium armor when they are a combat mage class. 
I'll close the PR, and open it when I get all the changes in for it to be ready for review

**Pull request closed without merging**